### PR TITLE
Add Compatibility

### PR DESCRIPTION
**Pull request by jerzzhang**
@gurpreetatwal I don't know if we need to add an additional README section on compatibility? I think I might add a guide to the Doc Center instead.

### CODE REVIEW

**Comment by KarthikBhaskara** [comment]
I don’t believe we need to add another read me on the sdk page itself. I mean as long as the route is documented? Doesn’t need to be a new read me right? 

Also add to api ref for sure. For doc center, probably just explaining exactly what compatibility means right now right?

File: README.md [hunk 1]
```diff
@@ -75,7 +75,7 @@ const client = new smartcar.AuthClient({
 // Redirect to Smartcar's authentication flow
 app.get('/login', function(req, res) {
 
-  const link = client.getAuthUrl({state: 'MY_STATE_PARAM'});
+  const link = client.getAuthUrl();
```
Comment (gurpreetatwal) [inline README.md]: why change this?

File: lib/auth-client.js [hunk 2]
```diff
@@ -202,4 +202,25 @@ AuthClient.prototype.exchangeRefreshToken = function(token) {
...
+ * @return {Promise.<Boolean>} true if the vehicle is compatible
+ * with Smartcar
+ */
+AuthClient.prototype.compatibility = function(vin) {
+
+  const qs = {vin};
+
+  return util.wrap(this.request.get({
+    baseUrl: config.api,
+    url: `${config.version}/compatibility`,
```
Comment (gurpreetatwal) [inline lib/auth-client.js]: missing `v` before version, code does not function as is :grimacing: 

```suggestion
    url: `v${config.version}/compatibility`,
```

File: test/unit/lib/auth-client.js [hunk 3]
```diff
@@ -342,3 +342,29 @@ test('exchangeRefreshToken', async function(t) {
...
+  const client = new AuthClient({
+    clientId: CLIENT_ID,
+    clientSecret: CLIENT_SECRET,
+    redirectUri: 'https://insurance.co/callback',
+  });
+
+  const vin = 'fake_vin';
+
+  const n = nock('https://api.smartcar.com')
+    .get('/1.0/compatibility')
```
Comment (gurpreetatwal) [inline test/unit/lib/auth-client.js]: missing `v` before version number

**Comment by gurpreetatwal** [comment]
I think guide in doc center would be better than modifying readme but we should update the sdk reference docs

**Comment by GregChan** [comment]
Make sure your final commit message uses semantic release conventions. Otherwise lgtm after you make Gurpreet’s changes.

File: README.md (hunk 1 above)
Comment (jerzzhang) [reply README.md]: @gurpreetatwal Jack was confused with this when building his first app because it was different than the Doc Center

Comment (gurpreetatwal) [reply README.md]: kk, ideally should be in a separate pr, don't want to conflate the changes

File: test/end-to-end/auth-client.js [hunk 4]
```diff
@@ -52,3 +52,11 @@ test('exchangeRefreshToken', async(t) => {
     'refreshToken',
   ]), []);
 });
+
+test('compatibility', async(t) => {
+  const client = new smartcar.AuthClient(getAuthClientParams());
+
+  const comp = client.compatibility('5YJXCDE22HF068739');
```
Comment (gurpreetatwal) [inline test/end-to-end/auth-client.js]: can you store the vin in a variable that describes the car, something like  `TESLA_MODEL_S` or something of the like

File: test/end-to-end/auth-client.js [hunk 5]
```diff
@@ -52,3 +52,11 @@ test('exchangeRefreshToken', async(t) => {
...
+
+  t.equal(comp, true);
+});
```
Comment (gurpreetatwal) [inline test/end-to-end/auth-client.js]: please add tests for false case as well

File: test/end-to-end/auth-client.js [hunk 6]
```diff
@@ -56,7 +56,12 @@ test('exchangeRefreshToken', async(t) => {
 test('compatibility', async(t) => {
   const client = new smartcar.AuthClient(getAuthClientParams());
 
-  const comp = client.compatibility('5YJXCDE22HF068739');
+  const teslaVin = '5YJXCDE22HF068739';
+  const royceVin = 'SCA665C59HUX86700';
 
-  t.equal(comp, true);
+  const teslaComp = client.compatibility(teslaVin);
+  const royceComp = client.compatibility(royceVin);
```
Comment (gurpreetatwal) [inline test/end-to-end/auth-client.js]: missing await?

File: lib/auth-client.js [hunk 7]
```diff
@@ -202,4 +202,25 @@ AuthClient.prototype.exchangeRefreshToken = function(token) {
...
+
+  const qs = {vin};
+
+  return util.wrap(this.request.get({
+    baseUrl: config.api,
+    url: `v${config.version}/compatibility`,
+    qs,
+  }))
+    .then(function(response) {
+      return response.compatibility;
```
Comment (gurpreetatwal) [inline lib/auth-client.js]: I should have noticed this earlier, but `compatible` not `compatibility`

**Pull request merged**